python/cudf/cudf/core/udf/utils.py

```diff
@@ -213,6 +225,37 @@ def _get_kernel(kernel_string, globals_, sig, func):
     globals_["f_"] = f_
     exec(kernel_string, globals_)
     _kernel = globals_["_kernel"]
-    kernel = cuda.jit(sig)(_kernel)
+    kernel = cuda.jit(sig, link=files, extensions=arg_handlers)(_kernel)
 
     return kernel
+
+
+launch_arg_getters: Dict[Any, Any] = {}
+
+
+def _get_input_args_from_frame(fr):
+    args = []
+    offsets = []
+    for col in _supported_cols_from_frame(fr).values():
+        getter = launch_arg_getters.get(col.dtype)
+        if getter:
+            data = getter(col)
+        else:
+            data = col.data
+        if col.mask is not None:
+            # argument is a tuple of data, mask
+            args.append((data, col.mask))
+        else:
+            # argument is just the data pointer
+            args.append(data)
+        offsets.append(col.offset)
+
+    return args + offsets
+
+
+def _return_col_from_dtype(dt, size):
```

Comment:
This name is misleading. It's returning a cupy array, not a column. Is that intentional/important?